lib/config.js

```diff
@@ -66,13 +66,14 @@ class Config {
         this.parser = options.parser;
         this.parserOptions = options.parserOptions || {};
 
+        this.configCache = new ConfigCache();
```

Comment:
Are we also supposed to flatten `extends` as well? Cause I'm getting 3 `extends` in baseConfig after resolution, one that was initially there, and 2 from the `array` config.